lib/Mojo/Base.pm

```diff
@@ -101,6 +105,16 @@ sub tap {
   return $self;
 }
 
+sub with_roles {
+  if ($_[0]->can_roles()) {
+    Carp::croak "class and roles must be package names" if !!grep { ref $_ } @_;
```

Comment:
OK, my error is nice, but if you try to invoke with_roles on an object, you get a terrifying exception, so I'm fine with that (something like "Can't locate Mojo/RoleTest/Base__WITH__Mojo/RoleTest/LOUD=HASH(0x17e56b0).pm in @INC", or whatever)